### PR TITLE
fix issues with validations, error handling and using correct bot user

### PR DESCRIPTION
**Pull request by darkLord19**
- If account is already disconnected or connected then return relevant message
- If user has already enabled or disabled activity notifications then send relevant messages
- Send file upload confirmation messages as plugin bot user instead of system user

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -136,7 +136,7 @@ ifneq ($(HAS_WEBAPP),)
 	mkdir -p dist/$(PLUGIN_ID)/webapp
 	cp -r webapp/dist dist/$(PLUGIN_ID)/webapp/
 endif
-	cd dist && tar -cvzf $(BUNDLE_NAME) $(PLUGIN_ID)
+	cd dist && COPYFILE_DISABLE=1 tar -cvzf $(BUNDLE_NAME) $(PLUGIN_ID)
```
Comment (fmartingr) [inline Makefile:139]: We tend not to deviate from our start template, why is this required?

**Review by fmartingr — CHANGES_REQUESTED** [review]
Made a small comment, can you please fill out the PR description as well for future reference? Thank you.

File: Makefile (hunk 1 above)
Comment (darkLord19) [reply Makefile:139]: `make deploy` fails in ARM Macbooks without this. There is open issue in mattermost repo https://github.com/mattermost/mattermost-server/issues/22614

Comment (fmartingr) [reply Makefile:139]: I recall having problems with tar myself. Did you try installing `gnutar` ?  https://formulae.brew.sh/formula/gnu-tar

File: server/plugin/disconnect.go [hunk 2]
```diff
@@ -6,7 +6,12 @@ import (
 )
 
 func (p *Plugin) handleDisconnect(c *plugin.Context, args *model.CommandArgs, _ []string) string {
-	err := p.client.KV.Delete(args.UserId + "_token")
+	var encryptedToken []byte
+	p.client.KV.Get(getUserTokenKey(args.UserId), &encryptedToken)
```
Comment (fmartingr) [inline server/plugin/disconnect.go:10]: Did you forget about this linter error?

**Comment by darkLord19** [comment]
@fmartingr addressed comments.

File: server/plugin/disconnect.go [hunk 3]
```diff
@@ -6,7 +6,12 @@ import (
 )
 
 func (p *Plugin) handleDisconnect(c *plugin.Context, args *model.CommandArgs, _ []string) string {
-	err := p.client.KV.Delete(args.UserId + "_token")
+	var encryptedToken []byte
+	_ = p.client.KV.Get(getUserTokenKey(args.UserId), &encryptedToken)
```
Comment (fmartingr) [inline server/plugin/disconnect.go:10]: What if there's a problem retrieving the item from the KVStore?

Comment (darkLord19) [reply server/plugin/disconnect.go:10]: This check is for checking if user is connected and then send appropriate message. if the get call fails for some reason then i think it's okay to let user proceed to actual disconnect operation and if there is any underlying issue with the server then delete operation will error out which is handled by sending appropriate message to user.

**Review by fmartingr — APPROVED** [review]

File: server/plugin/notifications.go [hunk 4]
```diff
@@ -289,8 +289,22 @@ func (p *Plugin) startDriveWatchChannel(userID string) error {
...
+func isWatchChannelDataValid(watchChannelData WatchChannelData) bool {
+	return watchChannelData.ChannelID != "" && watchChannelData.Expiration != 0 && watchChannelData.MMUserID != "" && watchChannelData.ResourceID != ""
+}
+
 func (p *Plugin) startDriveActivityNotifications(userID string) string {
-	err := p.startDriveWatchChannel(userID)
+	var watchChannelData WatchChannelData
+	err := p.client.KV.Get(getWatchChannelDataKey(userID), &watchChannelData)
+	if err != nil {
+		return "Something went wrong while starting Drive activity notifications. Please contact your organization admin for support."
```
Comment (wiggin77) [inline server/plugin/notifications.go:300]:
```suggestion
		return "Something went wrong while starting Google Drive activity notifications. Please contact your organization admin for support."
```

File: server/plugin/notifications.go [hunk 5]
```diff
@@ -289,8 +289,22 @@ func (p *Plugin) startDriveWatchChannel(userID string) error {
...
+	}
+
+	if isWatchChannelDataValid(watchChannelData) {
+		return "Drive activity notifications are already enabled for you."
```
Comment (wiggin77) [inline server/plugin/notifications.go:304]:
```suggestion
		return "Google Drive activity notifications are already enabled for you."
```

**Comment by wiggin77** [comment]
Looks great @darkLord19 .   There spots in the code where the integration is referred in user facing messages as "Drive".  Shouldn't this be "Google Drive"?

**Comment by darkLord19** [comment]
@wiggin77 yes making it Google Drive would be more self explanatory I think. Also should I rename the plugin bot user to google-drive because right now it's only drive. I wasnt sure what are the usual naming conventions for bot users.

**Comment by fmartingr** [comment]
> @wiggin77 @fmartingr can someone merge this PR? I am not getting the option to merge. I will raise separate PR to change the name of bot user and replace Drive with Google Drive in applicable places.

Done 👍
